server/src/main/java/org/opensearch/index/shard/IndexingStats.java

```diff
@@ -59,6 +62,90 @@ public class IndexingStats implements Writeable, ToXContentFragment {
      */
     public static class Stats implements Writeable, ToXContentFragment {
 
+        /**
+         * Tracks item level rest category class codes during indexing
+         *
+         * @opensearch.internal
+         */
+        public static class DocStatusStats implements Writeable, ToXContentFragment {
+
+            private final Map<String, AtomicLong> docStatusCounter;
```

Comment:
Do we need this map to maintain String as the key? Will it suffice to use an integer instead?

Reply:
I'm also in favor of keeping `Map<Integer, AtomicLong>` as per the perf numbers [here](https://github.com/opensearch-project/OpenSearch/pull/8716#issue-1807245528).